ENIGMAsystem/SHELL/Graphics_Systems/Direct3D11/DX11textures.cpp

```diff
@@ -65,8 +59,16 @@ int graphics_create_texture(const RawImage& img, bool mipmap, unsigned* fullwidt
   tdesc.CPUAccessFlags = 0;
   tdesc.MiscFlags = 0;
 
-  if (FAILED(m_device->CreateTexture2D(&tdesc,&tbsd,&tex)))
-    return 0;
+  if (img.pxdata != nullptr && (img.w != *fullwidth || img.h != *fullheight)) {
+    RawImage padded = image_pad(img, *fullwidth, *fullheight);
+    tbsd.pSysMem = padded.pxdata;
+    if (FAILED(m_device->CreateTexture2D(&tdesc,&tbsd,&tex)))
+      return 0;
+  } else {
+    tbsd.pSysMem = img.pxdata;
```

Comment:
While this is technically safe, this is also the null case for img.pxdata. Why not just return 0 early if that happens?

Reply:
You said, it's _also_ the null case, it's not _only_ the null case. This is where the already power of two textures get uploaded. For me to check that would be another nested if inside the else. I don't personally have a preference, but I know the function originally did not null check, as none of other graphics systems do. That's kind of undefined behavior here with respect to our mandatory API.

Reply:
It doesn't have to be inside the else. It can just be before the if: `if (!img.pxdata) return 0;`

It can be much before the if, even.

Reply:
What should we return -1 or 0 or something else? I think -1 is reasonable here, because it not only indicates error, but it's also safe since we use that for the default/no texture id.

Reply:
Nevermind, we can't do this because of the surface abstractions. I did this intentionally since surfaces are just render target textures. In other words, we create the surface texture by passing null pxdata.
https://github.com/enigma-dev/enigma-dev/blob/2d1220f23fa6dd057bf478eba8f25e5a82d34b6d/ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Desktop/surface.cpp#L30